Modules/FindCUDA/select_compute_arch.cmake

```diff
@@ -0,0 +1,168 @@
+# Synopsis:
+# Function for selecting GPU arch flags for nvcc based on CUDA_ARCH_NAME
+# Usage:
+#   SELECT_NVCC_ARCH_FLAGS(out_variable)
```

Comment:
I thought I mentioned this in some other place, but the documentation needs to be in FindCUDA.cmake, since that is where the auto-documentation generator will read it.  You should document the new functions and any variables produced by the script.  There are sections for each.